spec/command/base_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Command::Base do
+  subject(:command) { described_class.new(config) }
+
+  let(:config) { instance_double(Command::Config) }
+
+  around do |example|
+    suppress_output { example.run }
+  end
+
+  describe "#step" do
+    let(:message) { "test message" }
+
+    context "with retry_on_failure: true" do
+      let(:options) { { retry_on_failure: true, wait: 0 } }
+
+      it "does not exceed default max_retry_count" do
+        run_count = 0
+
+        command.step(message, **options) do
+          run_count += 1
+          false
+        end
+
+        expect(run_count).to eq 5
+      end
+
+      it "retries block until success" do
+        run_count = 0
+
+        command.step(message, **options) do
+          run_count += 1
+          true if run_count == 3
+        end
+
+        expect(run_count).to eq 3
+      end
+
+      context "with max_retry_count option" do
+        let(:options) { super().merge(max_retry_count: 1) }
```

Comment:
Because of `super()`, and also because we're shadowing `options`. This is much easier to read:

```
let(:options_with_max_retry_count) { options.merge(max_retry_count: 1) }
```

Or:

```
let(:options) { base_options.merge(max_retry_count: 1) }
let(:options) { common_options.merge(max_retry_count: 1) }
```

Reply:
Removed `super()` approach